pkg/infra/iac2/templates_compiler.go

```diff
@@ -398,7 +398,10 @@ func (tc TemplatesCompiler) handleIaCValue(v core.IaCValue) (string, error) {
 			return output, err
 		}
 		return output, nil
+	} else if reflect.TypeOf(v.Resource) == reflect.TypeOf((*resources.AvailabilityZones)(nil)) {
```

Comment:
```suggestion
	} else if _, ok := v.Resource.(*resources.AvailabilityZones); ok {
```
Use type assertion here, it's more idiomatic and performant.

Reply:
ill switch this over